### PR TITLE
Add cache interface for cn

### PR DESCRIPTION
**Pull request by nnsgmsone**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [X] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #3931 

## What this PR does / why we need it:

### CODE REVIEW

File: pkg/vm/engine/disttae/db.go [hunk 1]
```diff
@@ -15,9 +15,23 @@
...
 import (
+	"context"
+
 	"github.com/matrixorigin/matrixone/pkg/pb/plan"
+	"github.com/matrixorigin/matrixone/pkg/pb/timestamp"
+	"github.com/matrixorigin/matrixone/pkg/vm/engine"
 )
 
-func (db *DB) NewReader(databaseName, tableName string, expr *plan.Expr) Reader {
+func (db *DB) Update(ctx context.Context, databaseId, tableId uint64, ts timestamp.Timestamp) error {
```
Comment (ouyuanning) [inline pkg/vm/engine/disttae/db.go:25]: update what? only the timestamp?

Comment (nnsgmsone) [reply pkg/vm/engine/disttae/db.go:25]: Update the version of a table to the specified timestamp。

Comment (nnsgmsone) [reply pkg/vm/engine/disttae/db.go:25]: Do you need any other information than the timestamp?

Comment (ouyuanning) [reply pkg/vm/engine/disttae/db.go:25]: oh, when we get LogTailResponse,   we may need some interface to update DB's data

Comment (fengttt) [reply pkg/vm/engine/disttae/db.go:25]: Why this is a method of DB?   This *DB is shared by all sessions in CN?

File: pkg/vm/engine/disttae/types.go [hunk 2]
```diff
@@ -38,13 +41,26 @@ const (
...
 
-type Reader interface {
+// tae's block metadata, which is currently just an empty one, does not serve any purpose
+// When tae submits a concrete structure, it will replace this structure with tae's code
+type BlockMeta struct {
+}
+
+// Cache is a multi-version cache for maintaining some table data
+// The cache is concurrently secure, with multiple transactions accessing the cache at the same time
+type Cache interface {
```
Comment (fengttt) [inline pkg/vm/engine/disttae/types.go:57]: If there are 4 DNs, let's say, and your query touches only one, you poll only one.     Where is the infomation for this?   In context.Context?

**Review by fengttt — COMMENTED** [review]
More details in the API comment.

Line too long, can you break it to some readable length?

**Comment by nnsgmsone** [comment]
> More details in the API comment.
> 
> Line too long, can you break it to some readable length?

ok

File: pkg/vm/engine/disttae/db.go (hunk 1 above)
Comment (nnsgmsone) [reply pkg/vm/engine/disttae/db.go:25]: > Why this is a method of DB? This *DB is shared by all sessions in CN?
yes， cache is  multi-version, a cn has only one.

Comment (nnsgmsone) [reply pkg/vm/engine/disttae/db.go:25]: > yes， cache is multi-version, a cn has only one.

yes， cache is multi-version, a cn has only one.

**Review by reusee — APPROVED** [review]

**Pull request merged**